magellan/tests/test_attr_equiv_blocker.py

```diff
@@ -0,0 +1,230 @@
+        mg.set_key(A, l_key)
+        B = mg.read_csv_metadata(path_for_B)
+        mg.set_key(B, r_key)
+        ab = mg.AttrEquivalenceBlocker()
+        C = ab.block_tables(A, B, l_block_attr_1, r_block_attr_1, l_output_attrs, [])
+        s1 = ['_id', 'ltable_' + l_key, 'rtable_' + r_key]
+        s1 += ['ltable_' + x for x in l_output_attrs if x != l_key]
+        s1 = sorted(s1)
+        assert_equal(s1, sorted(C.columns))
+        assert_equal(mg.get_key(C), '_id')
+        assert_equal(mg.get_property(C, 'fk_ltable'), 'ltable_' + l_key)
+        assert_equal(mg.get_property(C, 'fk_rtable'), 'rtable_' + r_key)
+
+    @raises(AssertionError)
+    def test_ab_block_tables_bogus_l_block_attr(self):
+        A = mg.read_csv_metadata(path_for_A)
+        mg.set_key(A, l_key)
+        B = mg.read_csv_metadata(path_for_B)
+        mg.set_key(B, r_key)
+        ab = mg.AttrEquivalenceBlocker()
+        C = ab.block_tables(A, B, bogus_attr, r_block_attr_1)
+
+    @raises(AssertionError)
+    def test_ab_block_tables_bogus_r_block_attr(self):
+        A = mg.read_csv_metadata(path_for_A)
+        mg.set_key(A, l_key)
+        B = mg.read_csv_metadata(path_for_B)
+        mg.set_key(B, r_key)
+        ab = mg.AttrEquivalenceBlocker()
+        C = ab.block_tables(A, B, l_block_attr_1, bogus_attr)
+
+    @raises(AssertionError)
+    def test_ab_block_tables_multi_l_block_attr(self):
+        A = mg.read_csv_metadata(path_for_A)
+        mg.set_key(A, l_key)
+        B = mg.read_csv_metadata(path_for_B)
+        mg.set_key(B, r_key)
+        ab = mg.AttrEquivalenceBlocker()
+        C = ab.block_tables(A, B, block_attr_multi, r_block_attr_1)
+
+    @raises(AssertionError)
+    def test_ab_block_tables_multi_r_block_attr(self):
+        A = mg.read_csv_metadata(path_for_A)
+        mg.set_key(A, l_key)
+        B = mg.read_csv_metadata(path_for_B)
+        mg.set_key(B, r_key)
+        ab = mg.AttrEquivalenceBlocker()
+        C = ab.block_tables(A, B, l_block_attr_1, block_attr_multi)
+
+    @raises(AssertionError)
+    def test_ab_block_tables_bogus_l_output_attrs(self):
+        A = mg.read_csv_metadata(path_for_A)
+        mg.set_key(A, l_key)
+        B = mg.read_csv_metadata(path_for_B)
+        mg.set_key(B, r_key)
+        ab = mg.AttrEquivalenceBlocker()
+        C = ab.block_tables(A, B, l_block_attr_1, r_block_attr_1, [bogus_attr])
+
+    @raises(AssertionError)
+    def test_ab_block_tables_bogus_r_output_attrs(self):
+        A = mg.read_csv_metadata(path_for_A)
+        mg.set_key(A, l_key)
+        B = mg.read_csv_metadata(path_for_B)
+        mg.set_key(B, r_key)
+        ab = mg.AttrEquivalenceBlocker()
+        C = ab.block_tables(A, B, l_block_attr_1, r_block_attr_1, None, [bogus_attr])
+
+    def test_ab_block_candset(self):
+        A = mg.read_csv_metadata(path_for_A)
+        mg.set_key(A, l_key)
+        B = mg.read_csv_metadata(path_for_B)
+        mg.set_key(B, r_key)
+        ab = mg.AttrEquivalenceBlocker()
```

Comment:
All the tests seem to have these following lines,

A = mg.read_csv_metadata(path_for_A)
mg.set_key(A, l_key)
B = mg.read_csv_metadata(path_for_B)
mg.set_key(B, r_key)
ab = mg.AttrEquivalenceBlocker()

Can we put it in a "setup" function ?